### PR TITLE
[3.x] Make AnimationNodeBlendTree use `OrderedHashMap` insteads `Map`

### PR DESCRIPTION
**Pull request by TokageItLab**
3.x version of #79595. Supersedes #79531.

An ordered map must be used.

...I am not familiar with OrderedHashMap, does its order depend on the key? If it does not depend, then StringName is sufficient without making to be a String.

### CODE REVIEW

**Comment by AThousandShips** [comment]
You can use `StringName::AlphCompare` here as well, check the template for the map

**Comment by TokageItLab** [comment]
It is sufficient if the order of the inserted order, or the order of the index list of the map depending on the key, is unique.

At least from testing #79595 it seems to work well, but does unique comparison results mean that the order of the index list is unique? @AThousandShips

**Comment by AThousandShips** [comment]
Pointer comparison means unique order, in one run, and unless the names are freed and reallocated they'll remain ordered, the order doesn't persist between runs though

**Comment by TokageItLab** [comment]
Looking at the code for Map/RBMap, I see that the comparison by Comparator uniquely determines the index to be stored.

When StringName is used as a key of the Map/RBMap:
- When using AlphCompare, the index is uniquely determined by the string
- If nothing is specified, the index is determined by comparison of memory address, so if the memory address changes due to restart, etc., the index will change

@AThousandShips Is it correct?

**Comment by AThousandShips** [comment]
That sounds correct from my experience and reading of the code

My suggestion would be then:
* If the order between runs is important (like seems to be the case in 4.x) `AlphCompare` is good
* Otherwise pointer comparison should be preferred as it is *significantly* more performant

File: scene/animation/animation_blend_tree.cpp [hunk 1]
```diff
@@ -1121,7 +1121,7 @@ bool AnimationNodeBlendTree::_get(const StringName &p_name, Variant &r_ret) cons
 }
 void AnimationNodeBlendTree::_get_property_list(List<PropertyInfo> *p_list) const {
 	List<StringName> names;
-	for (Map<StringName, Node>::Element *E = nodes.front(); E; E = E->next()) {
+	for (NodeMap::Element *E = nodes.front(); E; E = E->next()) {
 		names.push_back(E->key());
 	}
 	names.sort_custom<StringName::AlphCompare>();
```
Comment (AThousandShips) [inline scene/animation/animation_blend_tree.cpp:1127]: This shouldn't be required any longer

File: scene/animation/animation_blend_tree.cpp [hunk 2]
```diff
@@ -870,7 +870,7 @@ Vector2 AnimationNodeBlendTree::get_node_position(const StringName &p_node) cons
 void AnimationNodeBlendTree::get_child_nodes(List<ChildNode> *r_child_nodes) {
 	Vector<StringName> ns;
 
-	for (Map<StringName, Node>::Element *E = nodes.front(); E; E = E->next()) {
+	for (NodeMap::Element *E = nodes.front(); E; E = E->next()) {
 		ns.push_back(E->key());
 	}
```
Comment (AThousandShips) [inline scene/animation/animation_blend_tree.cpp:876]: Or here (line below)

Comment (TokageItLab) [reply scene/animation/animation_blend_tree.cpp:876]: Applied same fix to master too.

File: scene/animation/animation_blend_tree.cpp (hunk 1 above)
Comment (TokageItLab) [reply scene/animation/animation_blend_tree.cpp]: Yes, already done.

**Review by AThousandShips — APPROVED** [review]
LGTM, haven't tested the code but usage is correct

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**